src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -1260,10 +1278,97 @@ internal bool CanAddParsedValue(HttpHeaderParser parser)
                 // supporting 1 value. When the first value gets parsed, CanAddValue returns true and we add the
                 // parsed value to ParsedValue. When the second value is parsed, CanAddValue returns false, because
                 // we have already a parsed value.
-                return parser.SupportsMultipleValues || ((InvalidValue == null) && (ParsedValue == null));
+                return parser.SupportsMultipleValues || !ContainsParsedValue();
```

Comment:
This logic seems like it's slightly different than the previous logic.

For "single-valued" headers, previously, we would allow either a single parsed value or a single invalid value, but not both. 

It seems like this is now allowing a single parsed value and an arbitrary number of invalid values. Is this intentional? If so what's the rationale for changing this?